src/ansys/sound/core/sound_composer/source_harmonics.py

```diff
@@ -0,0 +1,325 @@
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Sound Composer's harmonics source."""
+import warnings
+
+from ansys.dpf.core import Field, FieldsContainer, Operator
+from matplotlib import pyplot as plt
+import numpy as np
+
+from .._pyansys_sound import PyAnsysSoundException, PyAnsysSoundWarning
+from ._source_parent import SourceParent
+from .source_control_time import SourceControlTime
+
+ID_COMPUTE_LOAD_SOURCE_HARMONICS = "sound_composer_load_source_harmonics"
+ID_COMPUTE_GENERATE_SOUND_HARMONICS = "sound_composer_generate_sound_harmonics"
+
+
+class SourceHarmonics(SourceParent):
+    """Sound Composer's harmonics source class.
+
+    This class creates a harmonics source for the Sound Composer. A harmonics source is used to
+    generate a sound signal from a given harmonics source data and its source control. The
+    harmonics source data consists of a series of orders whose levels depend on RPM. The source
+    control contains the RPM values over time.
+    """
+
+    def __init__(self, file: str = "", source_control: SourceControlTime = None):
+        """Class instantiation takes the following parameters.
+
+        Parameters
+        ----------
+        file : str, default: ""
+            Path to the harmonics source data file. Supported files are the same XML and text (with
+            the header `AnsysSound_Orders`) formats as supported by Ansys Sound SAS.
+        source_control : SourceControlTime, default: None
+            Source control, consisting of the control parameter values over time, to use when
+            generating the sound from this source.
+        """
+        super().__init__()
+        self.source_control = source_control
+
+        # Define DPF Sound operators.
+        self.__operator_load = Operator(ID_COMPUTE_LOAD_SOURCE_HARMONICS)
+        self.__operator_generate = Operator(ID_COMPUTE_GENERATE_SOUND_HARMONICS)
+
+        if len(file) > 0:
+            self.load_source_harmonics(file)
+        else:
+            self.source_harmonics = None
+
+    def __str__(self) -> str:
+        """Return the string representation of the object."""
+        # Source info.
+        if self.source_harmonics is not None:
+            orders, control_name, control_values = self.__extract_harmonics_info()
+
+            # Source name.
+            str_name = self.source_harmonics.name
+            if str_name is None:
+                str_name = ""
+
+            # Orders.
+            orders = np.round(orders, 1)
```

Comment:
@anshlachamb This rounds to the first decimal. And it's only for the print display, the real data is not modified. I feel it makes sense for the context of print (even more so with arrays), in case you have weird values such as 3.0000000569 for example